dissect/target/plugins/os/windows/notifications.py

```diff
@@ -1,15 +1,150 @@
 from dissect.target.helpers.descriptor_extensions import UserRecordDescriptorExtension
-from dissect.target.helpers.record import create_extended_descriptor
+from dissect.target.helpers.record import RecordDescriptor, create_extended_descriptor
 from dissect.target.plugin import Plugin, export
 
+appdb_def = """
+typedef struct
+{
+  char     Magic[4];            // Always b"DNPW"
+  DWORD    Version;             // Versions 1 (win8) and 3 (win10) seem to exist
+  QWORD    Timestamp;           // According to some sources this is LastNotificationDate, but that seems incorrect.
+  DWORD    NextNotificationId;
+  DWORD    Unknown;
+  char     Padding[8];
+} ChunkHeader;                    // size: 0x20
+
+typedef struct
+{
+  WORD InUse;           // ??
+  WORD NextTileWrite;   // ??
+  WORD NextToastWrite;  // ??
+  BYTE Flags[2];
+} ChunkInfo;
+
+typedef struct
+{
+  QWORD    Timestamp1;              // ??
+  QWORD    Timestamp2;              // Is this time to refresh?
+  char     Uri[1024];               // Is this the correct size?
+  char     Padding[0x818 - 0x410];
+} PushDescriptor;                   // size: 0x818
+
+typedef struct
+{
+  DWORD    Id;
+  DWORD    Zero;              // ??
+  QWORD    Timestamp;         // ??
+  WORD     Unknown;
+  WORD     DataLength;
+  char     Data[DataLength];
+  char     Padding[0x118 - 0x14 - DataLength];
+} BadgeContent;  // size: 0x118
+
+typedef struct
+{
+  DWORD      UniqueId;        // ??
+  DWORD      Zero;
+  QWORD      ExpiryTime;      // The time this tile expires
+  QWORD      ArrivalTime;     // The time this tile was set
+  BYTE       Type;            // ??
+  BYTE       Index;
+  WORD       ContentLength;
+  wchar_t    Name[18];
+} TileDescriptor;             // size: 0x40
+
+typedef struct
+{
+  DWORD           UniqueId;        // ??
+  DWORD           Zero;
+  QWORD           ExpiryTime;      // The time this toast expires
+  QWORD           ArrivalTime;     // The time this toast was set
+  BYTE            Type;            // ??
+  BYTE            Index;
+  WORD            ContentLength;
+  wchar_t         Name1[17];
+  wchar_t         Name2[17];
+} ToastDescriptor;  // size: 0x60
+
+typedef struct
+{
+  char Content[0x1400];
+} DataXML;  // 0x1400
+
+typedef struct
+{
+  ChunkHeader     Header;            // Only populated for first chunk, else zeroed
+  ChunkInfo       Info;
+  PushDescriptor  Push;
+  BadgeContent    BadgeXml;
+  TileDescriptor  Tiles[5];          // start @ 0x958
+  DataXML         TileXml[5];
+
+  // For the in use chunks, 0x14 ToastDiscriptors have an Index, but there
+  // is space for more. Maybe this is used in case of deleted entries?
+  ToastDescriptor Toasts[0x14];      // start @ 0x6e98
+  char            Padding1[0x1e00];  // start @ 0x7618
+  DataXML         ToastXml[0x14];    // start @ 0x9418
+  char            Padding2[0x13f8];  // start @ 0x22418
+} Chunk;                             // size: 0x23810
```

Comment:
```suggestion
typedef struct {
    char    Magic[4];                   // Always b"DNPW"
    DWORD   Version;                    // Versions 1 (win8) and 3 (win10) seem to exist
    QWORD   Timestamp;                  // Some sources claim this is LastNotificationDate, but that seems incorrect
    DWORD   NextNotificationId;
    DWORD   Unknown;
    char    Padding[8];
} ChunkHeader;                          // size: 0x20

typedef struct {
    WORD    InUse;                      // ??
    WORD    NextTileWrite;              // ??
    WORD    NextToastWrite;             // ??
    BYTE    Flags[2];
} ChunkInfo;

typedef struct {
    QWORD   Timestamp1;                 // ??
    QWORD   Timestamp2;                 // Is this time to refresh?
    char    Uri[1024];                  // Is this the correct size?
    char    Padding[0x818 - 0x410];
} PushDescriptor;                       // size: 0x818

typedef struct {
    DWORD   Id;
    DWORD   Zero;                       // ??
    QWORD   Timestamp;                  // ??
    WORD    Unknown;
    WORD    DataLength;
    char    Data[DataLength];
    char    Padding[0x118 - 0x14 - DataLength];
} BadgeContent;                         // size: 0x118

typedef struct {
    DWORD   UniqueId;                   // ??
    DWORD   Zero;
    QWORD   ExpiryTime;                 // The time this tile expires
    QWORD   ArrivalTime;                // The time this tile was set
    BYTE    Type;                       // ??
    BYTE    Index;
    WORD    ContentLength;
    wchar_t Name[18];
} TileDescriptor;                       // size: 0x40

typedef struct {
    DWORD   UniqueId;                   // ??
    DWORD   Zero;
    QWORD   ExpiryTime;                 // The time this toast expires
    QWORD   ArrivalTime;                // The time this toast was set
    BYTE    Type;                       // ??
    BYTE    Index;
    WORD    ContentLength;
    wchar_t Name1[17];
    wchar_t Name2[17];
} ToastDescriptor;                      // size: 0x60

typedef struct {
    char    Content[0x1400];
} DataXML;                              // size: 0x1400

typedef struct {
    ChunkHeader     Header;             // Only populated for first chunk, else zeroed
    ChunkInfo       Info;
    PushDescriptor  Push;
    BadgeContent    BadgeXml;
    TileDescriptor  Tiles[5];           // start @ 0x958
    DataXML         TileXml[5];
    // For the in use chunks, 0x14 ToastDiscriptors have an Index, but there
    // is space for more. Maybe this is used in case of deleted entries?
    ToastDescriptor Toasts[0x14];       // start @ 0x6e98
    char            Padding1[0x1e00];   // start @ 0x7618
    DataXML         ToastXml[0x14];     // start @ 0x9418
    char            Padding2[0x13f8];   // start @ 0x22418
} Chunk;                                // size: 0x23810
```
Nitpicky alignment.

Reply:
Can you also still apply this?

Reply:
I thought I did, or did I miss something (modified it by hand)?

Reply:
I also indented the structures by 4 spaces, changed the location of the brackets (style choice), and the alignment of the final `//size:` comment.